src/types.ts

```diff
@@ -45,85 +36,88 @@ export type Group = {
         title?: string; // If not provided, will attempt to deduce.
       };
     }
+);
+
+// Object Endpoints Types
+export type GetObjectRequest = {
+  objectId: string;
+};
+
+export type ListObjectsRequest = {
+  parentId?: string;
+  limit?: number;
+  startingAfter?: string;
+  endingBefore?: string;
+};
+
+export type ListObjectsResponse = {
+  objects: Object[];
+  hasMore: boolean;
+};
+
+export type CreateObjectRequest = {
+  parentId?: string;
+  properties?: ObjectProperties;
+} & (
   | {
-      kind: 'email';
-      metadata: {
-        email: string;
-        sent?: Date; // If not provided, will attempt to deduce.
-        from?: string; // If not provided, will attempt to deduce.
-        subject?: string; // If not provided, will attempt to deduce.
-        to?: string[]; // If not provided, will attempt to deduce.
-      };
-    }
-  | {
-      kind: 'notion_page';
-      metadata: {
-        pageId: string;
-        url: string;
-        title?: string;
-      };
-    }
-  | {
-      kind: 'epub';
-      metadata: {
-        epubUrl: string;
-        title?: string;
-        language?: string;
-      };
+      type: 'collection';
+      metadata: {};
     }
   | {
-      kind: 'notion';
+      type: 'text';
       metadata: {
-        accessToken: string;
+        text: string;
       };
     }
   | {
-      kind: 'github_repository';
+      kind: 'html';
       metadata: {
-        accessToken: string;
-        repoOwner: string;
-        repoName: string;
-        rootPath: string;
+        html: string;
+        title?: string; // If not provided, will attempt to deduce.
       };
     }
   | {
-      kind: 'pdf';
+      kind: 'markdown';
       metadata: {
-        pdfUrl: string;
+        markdown: string;
+        title?: string; // If not provided, will attempt to deduce.
       };
     }
 );
 
+export type DeleteObjectRequest = {
+  objectId: string;
+};
+
+export type DeleteObjectResponse = {
+  deleted: boolean;
+};
+
+// Operation Endpoints Types
+
 export type Atom = {
   id: string;
   createdAt: Date;
   updatedAt: Date;
-  groupId: string;
+  objectId: string;
   content: string;
-  kind: 'title' | 'content' | 'link' | 'image' | 'code';
+  type: 'title' | 'content' | 'link' | 'image' | 'code' | 'list_item';
 };
 
-export type SearchResponse = {
-  id: string;
-  latencyMs: number;
-  atoms: (Atom & {
-    score: number;
-  })[];
-  groups: {
-    [groupId: string]: Group;
-  };
+export type SearchVariantContentsRequest = {
+  query: string;
+  parents?: string[];
+  max?: number;
+  filter?: Filter;
 };
 
-export type AskResponse = {
+export type SearchVariantContentsResponse = {
   id: string;
   latencyMs: number;
-  answer: string;
-  confidence: number;
-  sources: Group[];
+  contents: Atom[];
+  objects: Map<string, Object>;
 };
 
-export type CompletionResponse = {
-  latencyMs: number;
-  completions: string[];
-  sources: Group[];
+export type Filter = {
+  [key: string]: any;
```

Comment:
This should be fine for now, though in a future PR we can make this a little bit better typing.